### PR TITLE
feat(create-medusa-app,medusa-cli): Allow clearing project

### PR DESCRIPTION
**Pull request by shahednasser**
- Added build step
- Added `--no-boilerplate` option to `create-medusa-app` to allow clearing onboarding files
- Clear project files by default in medusa-cli

### CODE REVIEW

File: packages/create-medusa-app/src/index.ts [hunk 1]
```diff
@@ -6,6 +6,10 @@ program
   .description("Create a new Medusa project")
   .option("--repo-url <url>", "URL of repository to use to setup project.")
   .option("--seed", "Seed the created database with demo data.")
+  .option(
+    "--no-boilerplate",
+    "Install a Medusa project without the demo files."
```
Comment (olivermrbl) [inline packages/create-medusa-app/src/index.ts]:
```suggestion
    "Install a Medusa project without the boilerplate and demo files."
```

File: packages/create-medusa-app/src/utils/prepare-project.ts [hunk 2]
```diff
@@ -63,29 +69,76 @@ export default async ({
...
+  interval = createFactBox(spinner, "Running Migrations...", processManager)
+
   // run migrations
-  await runProcess({
+  await processManager.runProcess({
     process: async () => {
-      await promiseExec(
-        "npx -y @medusajs/medusa-cli@latest migrations run",
+      const proc = await promiseExec(
+        "npx -y @medusajs/medusa-cli@1.3.16-snapshot-20230613154819 migrations run",
```
Comment (olivermrbl) [inline packages/create-medusa-app/src/utils/prepare-project.ts]: todo: update

File: packages/create-medusa-app/src/utils/prepare-project.ts [hunk 3]
```diff
@@ -63,29 +69,76 @@ export default async ({
...
+      "Creating an admin user...",
+      processManager
+    )
 
-    await runProcess({
+    await processManager.runProcess({
       process: async () => {
         const proc = await promiseExec(
-          `npx -y @medusajs/medusa-cli@latest user -e ${admin.email} --invite`,
+          `npx -y @medusajs/medusa-cli@1.3.16-snapshot-20230613154819 user -e ${admin.email} --invite`,
```
Comment (olivermrbl) [inline packages/create-medusa-app/src/utils/prepare-project.ts]: todo: update

File: packages/create-medusa-app/src/utils/prepare-project.ts [hunk 4]
```diff
@@ -112,36 +170,41 @@ export default async ({
       return inviteToken
     }
 
-    await runProcess({
+    await processManager.runProcess({
       process: async () => {
         await promiseExec(
-          `npx -y @medusajs/medusa-cli@latest seed --seed-file=${path.join(
+          `npx -y @medusajs/medusa-cli@1.3.16-snapshot-20230613154819 seed --seed-file=${path.join(
```
Comment (olivermrbl) [inline packages/create-medusa-app/src/utils/prepare-project.ts]: todo: update

File: packages/create-medusa-app/src/utils/clear-project.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import fs from "fs"
```
Comment (olivermrbl) [inline packages/create-medusa-app/src/utils/clear-project.ts]: thought(non-blocking): Port to `@medusajs/utils` to re-use in `medusa-cli`

**Review by olivermrbl — APPROVED** [review]
LGTM - just want to test the snapshot before we merge, will report back

File: packages/create-medusa-app/src/utils/clone-repo.ts [hunk 6]
```diff
@@ -7,7 +7,8 @@ type CloneRepoOptions = {
 }
 
 // TODO change default repo URL
-const DEFAULT_REPO = "https://github.com/medusajs/medusa-starter-default"
+const DEFAULT_REPO =
```
Comment (shahednasser) [inline packages/create-medusa-app/src/utils/clone-repo.ts]: todo: update

**Comment by shahednasser** [comment]
@olivermrbl I created a snapshot now to test out using the `clearProject` function from `utils`. I'll make the changes based on the todos once we're done with the last snapshot 👍🏻

**Comment by shahednasser** [comment]
@olivermrbl the snapshot won't work properly since we removed the todos 😄

**Pull request merged**